appcues/src/main/java/com/appcues/data/remote/RetrofitWrapper.kt

```diff
@@ -0,0 +1,48 @@
+package com.appcues.data.remote
+
+import com.appcues.BuildConfig
+import com.appcues.data.MoshiConfiguration
+import okhttp3.HttpUrl
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import kotlin.reflect.KClass
+
+internal class RetrofitWrapper(
+    private val baseUrl: HttpUrl?,
+    private val isDebug: Boolean = BuildConfig.DEBUG,
+    private val interceptors: List<Interceptor> = listOf(),
+    private val okhttpConfig: (OkHttpClient.Builder) -> OkHttpClient.Builder = { it }
```

Comment:
this is mostly the same as before - when it was only used for appcues, but with enhancements
* `baseUrl` is optional
* you can pass in the custom list of interceptors based on the service
* you can chain on your own custom builder calls - like appcues uses to set the 5 sec timeout
* the retrofit instance is now a lazy prop, just in case multiple services are being created to use it (none do this currently)